src/Workspaces/Core/Portable/SolutionCrawler/InvocationReasons.cs

```diff
@@ -2,44 +2,39 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
-#nullable disable
-
 using System.Collections;
 using System.Collections.Generic;
 using System.Collections.Immutable;
+using System.Runtime.Serialization;
 
 namespace Microsoft.CodeAnalysis.SolutionCrawler
 {
+    [DataContract]
     internal partial struct InvocationReasons : IEnumerable<string>
```

Comment:
Move out of SolutionCrawler namespace to avoid dependency of the new interfaces on SolutionCrawler

Reply:
the next PR does that.